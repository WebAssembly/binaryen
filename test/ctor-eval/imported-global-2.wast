(module
  (memory 256 256)

  ;; imports must not be used
  (import "env" "imported" (global $imported i32))

  (func "test1" (result i32)
    (local $temp i32)

    ;; this errors, and we never get to evalling the store after it
    (local.set $temp
      (global.get $imported)
    )

    (i32.store8
      (i32.const 13)
      (i32.const 115)
    )

    (local.get $temp)
  )

  (func "keepalive" (result i32)
    (drop
      (i32.load
        (i32.const 13)
      )
    )
    (global.get $imported)
  )
)
