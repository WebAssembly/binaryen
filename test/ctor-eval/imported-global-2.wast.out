(module
 (type $none_=>_i32 (func (result i32)))
 (import "env" "imported" (global $imported i32))
 (memory $0 256 256)
 (export "test1" (func $0))
 (export "keepalive" (func $1))
 (func $0 (result i32)
  (local $temp i32)
  (local.set $temp
   (global.get $imported)
  )
  (i32.store8
   (i32.const 13)
   (i32.const 115)
  )
  (local.get $temp)
 )
 (func $1 (result i32)
  (drop
   (i32.load
    (i32.const 13)
   )
  )
  (global.get $imported)
 )
)
